src/components/modals/AddMovieModal/AddMovieModal.js

```diff
@@ -15,89 +19,147 @@ const FIELDS = [
     readOnly: true,
     key: 'id',
   },
-  { label: 'Title', type: 'text', key: 'title' },
-  { label: 'Release Date', type: 'date', key: 'release_date' },
-  { label: 'Poster Url', type: 'text', key: 'poster_path' },
+  {
+    label: 'Title',
+    type: 'text',
+    key: 'title',
+    placeholder: 'Input title',
+    validationSchema: Yup.string().required().label('Title'),
+  },
+  {
+    label: 'Release Date',
+    type: 'date',
+    key: 'release_date',
+    placeholder: 'Select Date',
+    validationSchema: Yup.date().nullable().label('Release Date'),
+  },
+  {
+    label: 'Poster Url',
+    type: 'text',
+    key: 'poster_path',
+    placeholder: 'Poster URL here',
+    validationSchema: Yup.string().url().required()
+      .label('Poster Url'),
+  },
   {
     label: 'Genre',
     type: 'select',
     values: GENRES,
     key: 'genres',
+    placeholder: 'Select Genre(s)',
+    validationSchema: Yup.array()
+      .of(Yup.string().oneOf(GENRES))
+      .min(1)
+      .nullable()
+      .label('Genres'),
+  },
+  {
+    label: 'Runtime',
+    type: 'number',
+    key: 'runtime',
+    placeholder: 'Runtime here',
+    validationSchema: Yup.number()
+      .required()
+      .positive()
+      .integer()
+      .label('Runtime'),
+  },
+  {
+    label: 'Overview',
+    type: 'textarea',
+    key: 'overview',
+    placeholder: 'Overview here',
+    validationSchema: Yup.string().required().label('Overview'),
   },
-  { label: 'Runtime', type: 'number', key: 'runtime' },
-  { label: 'Overview', type: 'textarea', key: 'overview' },
 ];
 
 const getOptions = (options) => (options ? options.map((option) => ({ label: option, value: option })) : []);
 
 const GENRES_OPTIONS = getOptions(GENRES.slice(0).sort(sortAbc));
 
-const AddMovieModal = ({
-  isEdit, movie, onAction, show,
-}) => {
-  const [mutableMovie, setMutableMovie] = useState({ ...movie });
+const validationSchemas = Object.assign(
+  {},
+  ...FIELDS.filter((field) => field.validationSchema).map((field) => ({
+    [field.key]: field.validationSchema,
+  })),
+);
 
-  useEffect(() => {
-    setMutableMovie({ ...movie });
-    return () => setMutableMovie({});
-  }, [show]);
+const commonValidationSchema = Yup.object(validationSchemas);
 
-  const onReset = () => {
-    setMutableMovie({ ...movie });
-  };
-
-  const onSaveOrSubmit = () => {
-    const actionName = isEdit ? 'update' : 'create';
-    onAction(actionName, mutableMovie);
-  };
+const MOVIE_DEFAULTS = {
+  title: '',
+  overview: '',
+  release_date: '',
+  poster_path: '',
+  runtime: '',
+  genres: null,
+};
 
-  const onInputChange = (fieldKey, type, newValue) => {
-    if (type === 'number') {
-      newValue = Number(newValue);
+const normalizeValues = (values) => {
```

Comment:
it's better make this function pure, cause right now it will mutate original values
```js
const normalizeValues = (values) => {
  const newValues = { ...values };
  Object.keys(newValues).forEach((key) => {
    if (isNullableOrEmpty(newValues[key])) {
      delete newValues[key];
    }
  });
  return newValues;
};
```

Reply:
Thanks, refactored